### ISSUE TITLE
Problems in switching to different datasets with iterator after restoring a pre-trained model for fine-tuning

### ISSUE THREAD

**Issue by jingmeng-bioinformatics** [tensorflow/tensorflow#19799]
Have I written custom code Yes
OS Platform and Distribution ubuntu 16.04 LTS
TensorFlow installed from pip3 install tensorflow-gpu
TensorFlow version the latest version
Bazel version NA
CUDA/cuDNN version CUDA Toolkit 9.0
GPU model and memory GeForce GTX 1080
Exact command to reproduce NA

I write a separate script to load the pre-trained model for fine-tuning with different datasets, then how do I get reference to the iterator and initialize it with fine_tune_dataset?

The following is a part of code in the script for training:

```
train_dataset = dataset_input_fn([‘/path/to/training_dataset’])
 validate_dataset = dataset_input_fn([‘/path/to/validation_dataset’])
 
 # create general iterator by the from_structure() method which needs the information of output data size/shape
 iterator = tf.data.Iterator.from_structure(train_dataset.output_types)
 data_batch, label_batch = iterator.get_next()

# make datasets that we can initialize seperately. 
 train_init_op = iterator.make_initializer(train_dataset)
 validate_init_op = iterator.make_initializer(validate_dataset)

accuracy, loss, train_op = model_function(data_batch, label_batch)

The following is the part of code in a separate script for restoring pre-trained model and fine_tune:

fine_tune_dataset = dataset_input_fn(['/path/to/fine_tune_model'])
validate_dataset = dataset_input_fn(['/path/to/validate_model'])
# make datasets that we can initialize seperately.
iterator = tf.data.Iterator.from_structure(fine_tune_dataset.output_types)
fine_tune_init_op = iterator.make_initializer(fine_tune_dataset)
validate_init_op = iterator.make_initializer(validate_dataset)
```
However, when I ran the script for fine-tuning, an error was given:

FailedPreconditionError (see above for traceback): GetNext() failed because the iterator has not been initialized. Ensure that you have run the initializer operation for this iterator before getting the next element.

I did call sess.run(fine_tune_init_op) in the fine-tuning part which did not show in the above. Could you please help me with how to switch to fine_tune_dataset after restoring pre-trained model in a different script? Thanks for your time in advance.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @angersson: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by angersson** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!
